### PR TITLE
misc cleanup

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: tools/public_api_guard/core/index.d.ts [hunk 1]
```diff
@@ -972,17 +972,17 @@ export declare class NoProviderError extends AbstractProviderError {
...
 /** @stable */
 export declare abstract class OnDestroy {
-    abstract ngOnDestroy(): any;
+    abstract ngOnDestroy(): void;
 }
 
 /** @stable */
 export declare abstract class OnInit {
-    abstract ngOnInit(): any;
+    abstract ngOnInit(): void;
```
Comment (DzmitryShylovich) [inline tools/public_api_guard/core/index.d.ts]: Finally :+1:

**Comment by vicb** [comment]
@kara could you please review this PR.

The main changes is about `trackBy` in `NgFor`: by using `onChange`, we make sure that the code is independent of input order - before it would fail if `ngForTrackBy` is listed after `ngFor` in the input list (or if `@Input` is added on the properties in the order they are laid out in the code).

**Comment by kara** [comment]
LGTM

**Comment by vicb** [comment]
Thanks for the review @kara

**Pull request merged**